File: src/main/java/saru/LadderMain.java
package saru;

import saru.domain.LadderGame;
import saru.view.LadderInput;
import saru.view.LadderOutput;

class LadderMain {
    private static final LadderInput ladderInput = new LadderInput();
    private static final LadderOutput ladderOutput = new LadderOutput();

    public static void main(String[] args) {
        String[] nameArr;
        String[] destination;
        int ladderHeight;

        do {
            nameArr = ladderInput.getUserName();
            destination = ladderInput.getDestination();
            ladderHeight = ladderInput.getHeight();
            ladderInput.flush();
        } while (!ladderInput.checkValid(nameArr, destination, ladderHeight));

        ladderOutput.sendOutputObject(new LadderGame(ladderHeight, destination, nameArr));
    }
}

File: src/main/kotlin/saru/LadderMain.kt
package saru

import saru.domain.LadderGame
import saru.view.LadderInput
import saru.view.LadderOutput

fun main() {
    val ladderInput = LadderInput()
    val ladderOutput = LadderOutput()

    var nameArr: Array<String>
    var destination: Array<String>
    var ladderHeight: Int

    do {
        nameArr = ladderInput.userName
        destination = ladderInput.destination
        ladderHeight = ladderInput.height
        ladderInput.flush()
    } while (!ladderInput.checkValid(nameArr, destination, ladderHeight))

    ladderOutput.sendOutputObject(LadderGame(ladderHeight, destination, nameArr))
}